### DISCUSSION TITLE
How to handle models with name having decimal?

### DISCUSSION THREAD

**Discussion by saadj55** [livewire/livewire#3518] [1. Help]
I have a faceted search to render on my page and the checkbox option values have decimal places.

Options values are rendered fine but whenever I try to select the checkbox, checksum exception is raised as below.

![image](https://user-images.githubusercontent.com/4143337/129334070-2b28cd46-47bb-46aa-b33f-b440b30703b8.png)

I even tried with sorted list of option but I am still facing this issue. I also tried to escape the decimal point as described here
 
[https://joelclermont.com/post/2020-09/understanding-dot-notation-in-laravel-validation/](url) but no luck. 
![image](https://user-images.githubusercontent.com/4143337/129334552-6c73118b-e69d-4805-8710-02ca67467b56.png)

Please help.

**Comment by ryatkins** [comment]
There could be two issues here:
1. The space in your name after Amperage and the parenthesis.
2. Livewire could be treating those decimal places and interpreting them to array dot notation

**Reply by saadj55** [reply]
I am sure its the 2nd issue here.

**Reply by ryatkins** [reply]
Can you paste in your Livewire component and Livewire view with Markdown styling?

**Reply by saadj55** [reply]
Correct me If I am wrong, I don't think Livewire has appropriate handling for this use case. 
The hydration middlewares do not account for this use case.

**Reply by saadj55** [reply]
Also, I tested, the space in the Amperage isn't an issue when working with whole numbers. As soon as I introduce decimal places, it breaks.

**Reply by ryatkins** [reply]
I would suggest creating an ID for your checkbox items and not trying to bind to the actual value of the checkbox.

Livewire supports the dots as model binding with array dot notation.

In my code base I'm using a deeply nested array below, but it could be simplified to 
`parts.{{ $index }}.property_of_model`

```
<input type="checkbox" name="bom_use_item_no" wire:model="parts.{{$form->form_id}}.{{$index}}.bom_use_item_no">
```

**Comment by joshhanley** [comment]
@saadj55 Livewire doesn't currently support properties with a dot in them, as @ryatkins suggested, it will be currently interpreting them as a part of the array dot notation.

I'm guessing the dot is because it is being used as part of a string key on an array, is that right?

**Reply by saadj55** [reply]
Yes, I am getting the product faceting from a meilisearch instance as a key value pair, hence I cannot associate every facet option and value to an ID as suggested by @ryatkins

**Reply by ryatkins** [reply]
@saadj55 can you string replace the dots to remove them or replace with another character?
